hoomd/pytest/test_simulation.py

```diff
@@ -0,0 +1,172 @@
+try:
+    import gsd.hoomd
+    skip_gsd = False
+except ImportError:
+    skip_gsd = True
+
+
+@pytest.fixture(scope="function")
+def get_snapshot(device):
+    def make_snapshot(n=10, particle_types=['A']):
+        s = hoomd.snapshot.Snapshot(device.comm)
+        if s.exists:
+            s.configuration.box = [20, 20, 20, 0, 0, 0]
+            s.particles.N = n
+            s.particles.position[:] = np.random.uniform(-10, 10, size=(n, 3))
+            s.particles.types = particle_types
+        return s
+    return make_snapshot
+
+
+def make_gsd_snapshot(hoomd_snapshot):
+    s = gsd.hoomd.Snapshot()
+    snap_properties = []
+    all_attr = dir(hoomd_snapshot)
+    for att in all_attr:
+        if att[0] != '_' and att not in ['exists', 'replicate']:
+            snap_properties.append(att)
+    for prop in snap_properties:
+        prop_attr = dir(getattr(hoomd_snapshot, prop))
+        nested_properties = []
+        for att in prop_attr:
+            if att[0] != '_':
+                nested_properties.append(att)
+        for nested_prop in nested_properties:
+            # s.prop.nested_prop = hoomd_snapshot.prop.nested_prop
+            setattr(getattr(s, prop), nested_prop,
+                    getattr(getattr(hoomd_snapshot, prop), nested_prop))
+    return s
+
+
+def set_types(s, inds, particle_types, particle_type):
+    for i in inds:
+        s.particles.typeid[i] = particle_types.index(particle_type)
+
+
+def update_positions(snap):
+    if snap.exists:
+        noise = 0.01
+        rs = np.random.RandomState(0)
+        mean = [0] * 3
+        var = noise * noise
+        cov = np.diag([var, var, var])
+        shape = snap.particles.position.shape
+        snap.particles.position[:] += rs.multivariate_normal(mean, cov,
+                                                             size=shape[:-1])
+    return snap
+
+
+def assert_equivalent_snapshots(snap1, snap2):
+    snap_properties = []
+    all_attr = dir(snap2)
+    for att in all_attr:
+        if att[0] != '_' and att not in ['exists', 'replicate']:
+            snap_properties.append(att)
+    for prop in snap_properties:
+        snap1_prop = getattr(snap1, prop)
+        snap2_prop = getattr(snap2, prop)
+        prop_attr = dir(snap2_prop)
+        nested_properties = []
+        for att in prop_attr:
+            if att[0] != '_':
+                nested_properties.append(att)
+        for nested_prop in nested_properties:
+            if nested_prop == 'types':
+                assert getattr(snap1_prop, nested_prop) == \
+                    getattr(snap2_prop, nested_prop)
+            else:
+                np.testing.assert_allclose(getattr(snap1_prop, nested_prop),
+                                           getattr(snap2_prop, nested_prop))
+
+
+def assert_equivalent_boxes(box1, box2):
+    metadata1 = box1.get_metadata()
+    metadata2 = box2.get_metadata()
+    for key in metadata1:
+        assert metadata1[key] == metadata2[key]
+
```

Comment:
This will break with the `feature/new-box-resize` PR #618. I would check their `L` and `tilts` directly.

Reply:
Fixed 055059e48